### DISCUSSION TITLE
Line spacing

### DISCUSSION THREAD

**Discussion by AlbertoFabbri93** [typst/typst#4542] [Q&A] [answered]
Hi,

I got a template from a friend where the line spacing is set to 2em with this line:

`#set par(leading: 2em,  justify: true, first-line-indent: 2em)`

It works well aside from bullet points where the line spacing between items is different:

![Screenshot 2024-07-11 at 17 10 32](https://github.com/typst/typst/assets/51910894/307df9b9-4579-4aa4-8c29-609e630e2cf1)

How do I set the line spacing between bullet points to be the same as between lines of a paragraph?

And I would like to restore the default line spacing inside tables, this is what I get right now:

![Screenshot 2024-07-11 at 17 13 34](https://github.com/typst/typst/assets/51910894/f9eb57ca-59f7-405b-9dca-1d3384ff6b6b)

I tried to play with `show` rules but unsuccessfully.

Thank you very much!

**Comment by eike** [comment, marked as answer]
You also need to set the spacing on the blocks created by paragraphs and list items, like so
```typst
#show par: set block(spacing: 2em)
#show list: set block(spacing: 2em)
```
(You might need to do this for additional elements, like `enum`, `heading`, etc.)

For the second question,
```typst
#show table: set par(leading: 0.65em)
```
should work. (0.65em is the default leading.)

**Comment by JosephBARBIERDARNAL** [comment]
For anyone finding this, we should now use this instead:

```typ
#set par(spacing: 2em)
```
